### PR TITLE
Issue#55 ts to babel to es5

### LINKED ISSUE #55: Add babel to the build process
We will add TS >> ES6 >> Babel >> ES5 step, by doing this we can get benefit of the more standard and granular way to handle polyfills.

### PR DESCRIPTION
**Pull request by crsanti**
This closes #55.

- Upgraded dependencies.
- Updated samples.

### CODE REVIEW

File: lib/package.json [hunk 1]
```diff
@@ -31,34 +31,33 @@
...
     "@types/karma-chai-sinon": "^0.1.5",
-    "@types/mocha": "^2.2.39",
-    "@types/sinon": "^1.16.31",
+    "@types/mocha": "^2.2.40",
+    "@types/sinon": "^1.16.35",
+    "awesome-typescript-loader": "^3.1.2",
+    "babel-core": "^6.24.0",
+    "babel-plugin-es6-promise": "^1.1.1",
+    "babel-preset-env": "^1.2.2",
+    "babel-preset-es2015": "^6.24.0",
```
Comment (nasdan) [inline lib/package.json]: Why do you need `babel-preset-env` and `babel-preset-es2015`?

File: lib/package.json [hunk 2]
```diff
@@ -31,34 +31,33 @@
...
     "@types/chai-as-promised": "^0.0.30",
-    "@types/core-js": "0.9.36",
     "@types/karma-chai-sinon": "^0.1.5",
-    "@types/mocha": "^2.2.39",
-    "@types/sinon": "^1.16.31",
+    "@types/mocha": "^2.2.40",
+    "@types/sinon": "^1.16.35",
+    "awesome-typescript-loader": "^3.1.2",
+    "babel-core": "^6.24.0",
+    "babel-plugin-es6-promise": "^1.1.1",
```
Comment (nasdan) [inline lib/package.json]: Why do you need this package?

File: lib/webpack.config.js [hunk 3]
```diff
@@ -5,20 +5,16 @@ var CopyWebpackPlugin = require('copy-webpack-plugin');
...
-    // .js is required for react imports.
-    // .tsx is for our app entry point.
-    // .ts is optional, in case you will be importing any regular ts files.
-    extensions: ['', '.js', '.ts', '.tsx']
+    extensions: ['.js', '.ts']
   },
 
   entry: [
-    './index.ts'
+    'es6-promise',
```
Comment (nasdan) [inline lib/webpack.config.js:16]: Is it worth if we move `es6-promise` to vendors?

File: lib/webpack.config.js [hunk 4]
```diff
@@ -28,38 +24,28 @@ var config = {
   },
 
   module: {
-    loaders: [
+    rules: [
       {
         test: /\.(ts|tsx)$/,
         exclude: /node_modules/,
-        loader: 'ts-loader'
+        use: [
```
Comment (nasdan) [inline lib/webpack.config.js]: Array is not necessary

File: samples/react/00 SimpleForm/webpack.config.js [hunk 5]
```diff
@@ -53,28 +53,39 @@ module.exports = {
...
+      { test: /\.eot(\?v=\d+\.\d+\.\d+)?$/, loader: "file-loader" },
+      { test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: "url-loader?limit=10000&mimetype=image/svg+xml" }
     ]
   },
 
   plugins: [
-    new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.js'),
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      filaneme: 'vendor.js'
```
Comment (nasdan) [inline samples/react/00 SimpleForm/webpack.config.js]: typo

Comment (nasdan) [reply samples/react/00 SimpleForm/webpack.config.js]: Btw remove line, it's not necessary

File: samples/react/01 SignupForm/webpack.config.js [hunk 6]
```diff
@@ -54,28 +53,39 @@ module.exports = {
...
+      { test: /\.eot(\?v=\d+\.\d+\.\d+)?$/, loader: "file-loader" },
+      { test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: "url-loader?limit=10000&mimetype=image/svg+xml" }
     ]
   },
 
   plugins: [
-    new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.js'),
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      filaneme: 'vendor.js'
```
Comment (nasdan) [inline samples/react/01 SignupForm/webpack.config.js]: Remove this line

File: samples/react/02 QuizForm/webpack.config.js [hunk 7]
```diff
@@ -55,28 +55,39 @@ module.exports = {
...
+      { test: /\.eot(\?v=\d+\.\d+\.\d+)?$/, loader: "file-loader" },
+      { test: /\.svg(\?v=\d+\.\d+\.\d+)?$/, loader: "url-loader?limit=10000&mimetype=image/svg+xml" }
     ]
   },
 
   plugins: [
-    new webpack.optimize.CommonsChunkPlugin('vendor', 'vendor.js'),
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      filaneme: 'vendor.js'
```
Comment (nasdan) [inline samples/react/02 QuizForm/webpack.config.js]: Remove this line

File: lib/package.json (hunk 1 above)
Comment (crsanti) [reply lib/package.json]: `babel-preset-env` to take care of target. Including `babel-preset-es2015` preset was a mistake.

File: lib/package.json (hunk 2 above)
Comment (crsanti) [reply lib/package.json]: I did some tries to optimize the dist package size. I forgot to remove this package...

File: lib/webpack.config.js (hunk 4 above)
Comment (crsanti) [reply lib/webpack.config.js]: Ok! Changing it~!

**Review by nasdan — APPROVED** [review]

File: lib/webpack.config.js (hunk 3 above)
Comment (crsanti) [reply lib/webpack.config.js:16]: Since this ibrary will be transpiled as UDMD I don't think the advantage of this.

**Pull request merged**